src/models/beast.ts

```diff
@@ -16,20 +24,136 @@ export default class BeastModel {
+            BeastModel.handleErrors(error);
+        }
+        // this won't ever get hit, but eslint doesn't like when its not here for some reason lol
+        throw new UnknownError('An unexpected error occurred');
+    }
+
+    /**
+     * Get a beast by its id, also includes the beast's friends and preeshes made/received
+     * @param id Id of the requested beast
+     */
+    public static async getBeastById(id: number): Promise<Beast | null> {
+        try {
+            const beast = await prisma.beast.findUnique({
+                where: {
+                    id,
+                },
+                include: {
+                    beastieBros: true,
+                    authoredPreesh: {
+                        include: {
+                            comments: true,
+                        },
+                    },
+                    receivedPreesh: {
+                        include: {
+                            comments: true,
+                        },
+                    },
+                },
+            });
+            if (!beast) {
+                throw new NotFoundError(`Beast with id: ${id} does not exist.`);
+            }
+
+            return beast;
+        } catch (error: unknown) {
+            BeastModel.handleErrors(error);
+        }
+        throw new UnknownError('An unexpected error occurred');
+    }
+
+    /**
+     * Get a beast's friends by its id
+     * @param id The id of the beast
+     */
+    public static async getBeastFriends(id: number): Promise<Beast[] | null> {
+        try {
+            const beast = await prisma.beast.findUnique({
+                where: {
+                    id,
+                },
+                include: {
+                    beastieBros: true,
+                },
+            });
+            if (!beast) {
+                throw new NotFoundError(`Beast with id: ${id} does not exist.`);
+            }
+
+            return beast.beastieBros;
+        } catch (error: unknown) {
+            this.handleErrors(error);
+        }
+        throw new UnknownError('An unexpected error occurred');
+    }
+
+    /**
+     * Updates a beast with the given (optional) gamerTag and (optional) email
+     * @param id the id of the beast to update
+     * @param gamerTag the new gamerTag
+     * @param email the new email
+     */
+    public static async updateBeast(id: number, gamerTag?: string, email?: string): Promise<Beast | null> {
+        // Will only update with the fields that are provided
+        const updateData: { gamerTag?: string; email?: string; } = {};
+
+        if (gamerTag) {
+            updateData.gamerTag = gamerTag;
+        }
+
+        if (email) {
+            updateData.email = email;
+        }
+
+        // Short circuit if there is nothing to update
+        // tbh not really sure what error should be thrown here but def short circuit before accessing db
+        // so just doing this xD :P :3
```

Comment:
Can define and throw an IllegalArgument exception, would probably be good to have for the future

Reply:
i think this can be removed since it should be handled in a parameter validation